erc20/src/lib.rs

```diff
@@ -29,15 +29,29 @@ impl ERC20 {
 
         self.balance.write(from, from_balance - value);
         self.balance.write(to, to_balance + value);
+
+        let mut log_data = [0u8; 48];  // 20 + 20 + 8 bytes
+        log_data[..20].copy_from_slice(from.as_slice());
+        log_data[20..40].copy_from_slice(to.as_slice());
+        log_data[40..48].copy_from_slice(&value.to_ne_bytes());
+    
+        log(
+            log_data.as_ptr() as u64,
+            log_data.len() as u64,
+            0,
+            0
+        );
```

Comment:
Can you extract these into a generic function so that here you could just make the call below? Internally the `log` function needs to ABI encode the values and use the signature of the string, like with function selectors (to keep compatibility with Solidity events).

```rust
log("LogName", from, to, value);
```